basyx/aas/adapter/http.py

```diff
@@ -100,49 +100,63 @@ class StrippedResultToJsonEncoder(ResultToJsonEncoder):
 
 class APIResponse(abc.ABC, Response):
     @abc.abstractmethod
-    def __init__(self, obj: Optional[ResponseData] = None, stripped: bool = False, *args, **kwargs):
+    def __init__(self, obj: Optional[ResponseData] = None, cursor: Optional[int] = None,
+                 stripped: bool = False, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if obj is None:
             self.status_code = 204
         else:
-            self.data = self.serialize(obj, stripped)
+            self.data = self.serialize(obj, cursor, stripped)
 
     @abc.abstractmethod
-    def serialize(self, obj: ResponseData, stripped: bool) -> str:
+    def serialize(self, obj: ResponseData, cursor: int, stripped: bool) -> str:
```

Comment:
`obj: Optional[ResponseData] = None`. ResponseData can be None in the initilization as well right? Why can't it be None when we serialize?

Reply:
Because `__init__` checks whether `obj is None` and doesn't call `serialize()` in this case. Thus, `serialize()` is only called if `obj` is not `None`.

Reply:
I get it, Thanks :D